internal/bminventory/inventory.go

```diff
@@ -2674,24 +2674,42 @@ func (b *bareMetalInventory) handleReplyError(params installer.V2PostStepReplyPa
 }
 
 func (b *bareMetalInventory) handleMediaDisconnection(params installer.V2PostStepReplyParams, ctx context.Context, log logrus.FieldLogger, h *models.Host) error {
-	status := models.HostStatusError
 	statusInfo := fmt.Sprintf("%s - %s", string(models.HostStageFailed), mediaDisconnectionMessage)
 
+	if params.Reply.Error != "" {
+		statusInfo = fmt.Sprintf("%s. %s", statusInfo, params.Reply.Error)
+	}
+
 	// Install command reports its status with a different API, directly from the assisted-installer.
 	// Just adding our diagnose to the existing error message.
-	if swag.StringValue(h.Status) == status && h.StatusInfo != nil {
+	if swag.StringValue(h.Status) == models.HostStatusError {
 		// Add the message only once
-		if strings.Contains(*h.StatusInfo, statusInfo) {
+		if h.StatusInfo != nil && strings.Contains(*h.StatusInfo, statusInfo) {
 			return nil
 		}
 
-		statusInfo = fmt.Sprintf("%s. %s", statusInfo, *h.StatusInfo)
-	} else if params.Reply.Error != "" {
-		statusInfo = fmt.Sprintf("%s. %s", statusInfo, params.Reply.Error)
+		if h.StatusInfo != nil {
+			statusInfo = fmt.Sprintf("%s. %s", statusInfo, *h.StatusInfo)
+		}
+
+		// Update status info
+		err := b.db.Model(h).Updates(map[string]interface{}{
+			"status_info": statusInfo,
+		}).Error
+
+		if err != nil {
+			return err
+		}
+
+		// Create event even for the status info change even when the status remain the same
```

Comment:
it means that we will get 2 events:
one event - host status was changed to error without status info and in another one, the host status was changed from error to error with status info, right?
Do we use this event in our data dashboards in elastic? want to be sure that it won't affect our reports
Another question - I hope that this behavior is only for the short term and we won't have a duplicate event in the better solution.

Reply:
In the Elastic side we aggregate the events by cluster id.
Un the next version we should update the event.
Maybe even for the current version let me explore it

Reply:
The solution should be creating a dedicated media disconnection event. Having fake event isn't good idea (having a status change event without changing the status) 
Updating the existing event isn't good solution too since it will break every consumer. Event should be unmutable